Objects/regions.c

```diff
@@ -802,17 +809,6 @@ static int PyRegion_init(PyRegionObject *self, PyObject *args, PyObject *kwds) {
     Py_SET_REGION(self, (Py_uintptr_t) self->metadata);
     _Py_MakeImmutable(_PyObject_CAST(Py_TYPE(self)));
 
-    // FIXME: Usually this is created on the fly. We need to do it manually to
-    // set the region and freeze the type
-    self->dict = PyDict_New();
-    if (self->dict == NULL) {
-        return -1; // Propagate memory allocation failure
-    }
-    // TODO: Once phase 2 is done, we might be able to do this statically
-    // at compile time.
-    _Py_MakeImmutable(_PyObject_CAST(Py_TYPE(self->dict)));
```

Comment:
The tests currently manually freeze the dictionary type. I feel like this will be a common pitfall when crafting examples.

@matajoh Is there a startup function or another way we could freeze the dictionary type on startup?

Reply:
Is this a case for `makeglobalsimmutable()`?

Reply:
It depends on the semantics of `makeglobalsimmutable()`. Thus far, I've heard the rule that it has to be called before we start having behaviors / threads work on regions. While we would have to make the dictionary type immutable when we create the first regions.